src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -93,15 +96,20 @@ protected override VisualStudioWorkspaceImpl CreateWorkspace()
 
         protected override void Dispose(bool disposing)
         {
-            UnregisterObjectBrowserLibraryManager();
+            if (disposing)
+            {
+                JoinableTaskFactory.Run(() => UnregisterObjectBrowserLibraryManagerAsync(CancellationToken.None));
```

Comment:
More a general curiosity: what is the relative ordering of Dispose() running and the InitializeAsync() cancellation being observed in the case of a shutdown? Could this Dispose() run while RegisterObjectBrowserLibraryManagerAsync is waiting for the service?

Reply:
That's a great question for @ryanmolden: does the package manager guarantee to not call `Dispose()` before the package's `InitializeAsync` Task has completed?

Reply:
so this is called from UI thread or BG thread?

Reply:
so if there is InitializeAsync that is called from BG, async, shouldn't there be corresponding DisposeAsync that runs from BG, async?

Reply:
> so this is called from UI thread or BG thread?

I'm not sure. I'm assuming that setting `AllowsBackgroundLoading` doesn't change whatever it was doing before.

> so if there is InitializeAsync that is called from BG, async, shouldn't there be corresponding DisposeAsync that runs from BG, async?

There is no such method. This would be a question for the VS platform team.

Reply:
@AArnott @sharwell @ryanmolden did we ever confirm the behavior here?